### PR TITLE
Bug 1800654 - don't try to add a profile not provided, or without a name

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Use slice of ptrs for GetContainerProfileInfo and ensure no attempt to write profile with empty name.

## QA steps

1. vim ./testcharms/charm-repo/quantal/lxd-profile-alt/lxd-profile.yaml, remove the Devices section.
2. juju bootstrap localhost
2.  juju deploy ubuntu --to lxd  -n 2
3. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-alt --to lxd -n 2

All 4 units should succeed, the lxd-profile-alt ones should have charm profiles attached.

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1800654

### CODE REVIEW

File: apiserver/facades/client/client/status.go [hunk 1]
```diff
@@ -915,7 +915,7 @@ func (c *statusContext) makeMachineStatus(machine *state.Machine, appStatusInfo
 			}
 		}
 	} else {
-		logger.Debugf("error fetching lxd profiles for %s: %q", machine.String(), err.Error())
+		logger.Tracef("error fetching lxd profiles for %s: %q", machine.String(), err.Error())
```
Comment (hmlanigan) [inline apiserver/facades/client/client/status.go]: drive by

File: worker/provisioner/provisioner_task.go [hunk 2]
```diff
@@ -377,20 +377,16 @@ func (task *provisionerTask) processOneMachineProfileChange(
 	logger.Debugf("processOneMachineProfileChange(%s)", m.Id())
 	info, err := m.CharmProfileChangeInfo()
 	if err != nil {
-		logger.Debugf("processOneMachineProfileChange(%s) CharmProfileChangeInfo failed with %v", m.Id(), err)
```
Comment (hmlanigan) [inline worker/provisioner/provisioner_task.go]: changes in the file a drive by, should have been removed before landing

**Comment by hmlanigan** [comment]
[cannot write charm profile: No name provided](https://trello.com/c/JYMyLpH2/741-cannot-write-charm-profile-no-name-provided)

File: api/provisioner/provisioner.go [hunk 3]
```diff
@@ -407,27 +407,32 @@ func (a *State) CACert() (string, error) {
...
 	profiles := result.Results[0].LXDProfiles
-	res := make([]LXDProfileResult, len(profiles))
-	for i, p := range profiles {
-		res[i].Config = p.Profile.Config
-		res[i].Description = p.Profile.Description
-		res[i].Devices = p.Profile.Devices
-		res[i].Name = p.Name
+	var res []*LXDProfileResult
+	for _, p := range profiles {
+		if p == nil {
```
Comment (SimonRichardson) [inline api/provisioner/provisioner.go]: :+1:

**Comment by SimonRichardson** [comment]
QA done.

**Comment by SimonRichardson** [comment]
!!build!!

> Intermittent failure

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**